content/blog/egress-traffic-with-kgateway-and-Istio-integration.md

```diff
@@ -0,0 +1,291 @@
+
+This separation lets platform teams choose when L7 processing is necessary, reducing cost and computational overhead for workloads that only require transport security.
+
+## kgateway's integration with Ambient Mesh
+kgateway integrates to Ambient Mesh for managing our workloads through Layer 4 and Layer 7 network policies. But the thing that sets its apart from other Gateway solutions is that, Kgateway is the first project that can be used as a pluggable waypoint for Istio. 
+Kgateway has been built on same Envoy engine that Istio’s waypoint implementation uses, which has certain features including Istio API Compatability, Shared Observability, Faster Adoption of Security Featrues and Unified Configurational Model with Ambient Mesh.
+
+# How important is kgateway's integration
+Before integrating kagteway with Istio Ambient, ensure we have: 
+1. Set-up `kind` cluster.
+2. Setup Kuberntes Gateway API:
+   ```
+   kubectl apply --server-side -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.4.0/experimental-install.yaml
+   ```
+4. Follow the [Get started guide](https://kgateway.dev/docs/latest/quickstart/) to install kgateway.
+5. Follow the [Sample app guide](https://kgateway.dev/docs/latest/install/sample-app/) to create a gateway proxy with an HTTP listener and deploy the httpbin sample app.
+6. Set up an ambient mesh in your cluster to secure service-to-service communication with mutual TLS by following the [ambientmesh.io](https://ambientmesh.io/docs/quickstart/) quickstart documentation.
+7. Deploy the Ollama Container at port number 11434, binding to 0.0.0.0 so the Kubernetes virtual machine can access it via the host's bridge network.
+   ```
+   docker run -d -v ollama:/root/.ollama -p 11434:11434 -e OLLAMA_HOST=0.0.0.0 ollama/ollama --name ollama-server
+   ```
+8. Get the Container IP of ollama which will be inserted at all the **`address filds` which is 172.17.0.2 in our case.
+   ```
+   docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' ollama-server
+   ```
+9. Deploy the Kyverno Authz Server is a GRPC server capable of processing Envoy External Authorization requests.
+```
+helm install kyverno-authz-server --namespace default --create-namespace --wait \
+  --version 0.1.0 --repo https://kyverno.github.io/kyverno-envoy-plugin \
+  kyverno-authz-server
+```
+
+
+While Istio ambient provides Authorization, Authentication and Egress still, there are several scenarios where kgateway can offer a more powerful alternative:
+
+## Securely Egress Traffic with kGateway + Istio Integration
+
+To establish a dedicated, policy-enforced egress path, we must combine three core resources: the **Istio ServiceEntry** (to register the external host), the **kGateway Egress Gateway** (to serve as the L7 egress waypoint), and the **HTTPRoute** (to apply the routing and security logic).
```

Comment:
I think we should maybe remove the word waypoint from here, because it is not using a waypoint gateway class